### PR TITLE
0145 b clip currante rate if out of bound

### LINKED ISSUE #144: Calculate actual exchange rate in FileUpdate memo
**Summary**
We currently include the current, next and midnight values for hbarEquiv in the memo of the hourly FileUpdate tx. 

It would simplify analysis if those values were divided by 30k to give the actual rates

### PR DESCRIPTION
**Pull request by anighanta**
**Related issue(s)**:
Closes #144 #145 #148 

**Summary of the change**:
We are now comparing currentRate as well with the `currentRate` in `midnightRate` and check if is in bound. 
If not we are clipping it as well.

Updated memo field to show the rate in `USD` instead.

Updated `jackson-databind` dependency to the latest version from maven.

**Applicable documentation**
- [x] Javadoc
- [x] tests
- [x] README

### CODE REVIEW

File: src/main/java/com/hedera/exchange/ERTproc.java [hunk 1]
```diff
@@ -139,10 +139,17 @@ public ExchangeRate call() {
...
+            if(midnightExchangeRate != null) {
+                if(!midnightExchangeRate.getNextRate().isSmallChange(this.bound, nextRate)) {
+                    LOGGER.debug(Exchange.EXCHANGE_FILTER, "last midnight value present. Validating the nextRate with {}",
+                            midnightExchangeRate.getNextRate().toJson());
+                    nextRate = midnightExchangeRate.getNextRate().clipRate(nextRate, this.bound);
+                }
+                if(!midnightExchangeRate.getCurrentRate().isSmallChange(this.bound, currentExchangeRate)) {
+                    LOGGER.debug(Exchange.EXCHANGE_FILTER, "last midnight value present. Validating the currentRate with {}",
+                            midnightExchangeRate.getCurrentRate().toJson());
+                    currentExchangeRate = midnightExchangeRate.getCurrentRate().clipRate(currentExchangeRate, this.bound);
```
Comment (cesaragv) [inline src/main/java/com/hedera/exchange/ERTproc.java:151]: wouldn't this one already have been clipped? By the previous run?

File: src/main/java/com/hedera/exchange/ExchangeRateTool.java [hunk 2]
```diff
@@ -165,7 +164,7 @@ private static void execute() throws Exception {
     private static void updateTransactionFileForNetwork(final String networkName,
             final AccountId operatorId,
             final ExchangeRate exchangeRate,
-            final Rate midnightRate,
+            final ExchangeRate midnightRate,
```
Comment (cesaragv) [inline src/main/java/com/hedera/exchange/ExchangeRateTool.java:167]: should be renamed to `midnightExchangeRate`

File: src/main/java/com/hedera/exchange/Rate.java [hunk 3]
```diff
@@ -249,4 +249,12 @@ public Rate clipRate(final Rate newRate, long bound) {
     public String toJson() throws JsonProcessingException {
         return OBJECT_MAPPER.writeValueAsString(this);
     }
+
+    @JsonIgnore
+    /**
+     * Get the hbar rate in USD
+     */
+    public double getRateinUSD() {
```
Comment (cesaragv) [inline src/main/java/com/hedera/exchange/Rate.java:257]: This method doesn't appear to have unit tests

**Review by cesaragv — CHANGES_REQUESTED** [review]
I don't see SonarQube running on this branch

File: src/main/java/com/hedera/exchange/ERTproc.java (hunk 1 above)
Comment (anighanta) [reply src/main/java/com/hedera/exchange/ERTproc.java:151]: yes, in the previous Run it is clipped when compared to nextRate of MidnightExchangeRate, now it is clipped by comparing it to currentRate of MidnightExchangeRate

File: src/main/java/com/hedera/exchange/ExchangeRateTool.java (hunk 2 above)
Comment (anighanta) [reply src/main/java/com/hedera/exchange/ExchangeRateTool.java:167]: agreed

File: src/main/java/com/hedera/exchange/Rate.java (hunk 3 above)
Comment (anighanta) [reply src/main/java/com/hedera/exchange/Rate.java:257]: Done

Comment (cesaragv) [inline src/main/java/com/hedera/exchange/Rate.java:257]: it should be `getRateInUSD`

**Review by cesaragv — APPROVED** [review]

**Pull request merged**